Source/Python/application/requests/rest/handlers.py

```diff
@@ -0,0 +1,342 @@
+        def location_patch_handler(body_patch_operations: List[Dict[str, Any]], *, location_id: str, **kwargs) -> HandlerResult:
+            if not location_id.startswith("location-"):
+                raise ValueError(f"Cannot parse location id from '{location_id}")
+            location_id_ = JsonTranslator.from_json(location_id, PrefixedUUID)
+
+            patch = JsonPatch([PatchOperation(operation).operation for operation in body_patch_operations])
+            existing_location_json = JsonTranslator.to_json(self._location_use_case.retrieve(location_id_, **kwargs))
+
+            modified_location_json = patch.apply(existing_location_json)
+            modified_location = JsonTranslator.from_json(modified_location_json, Location)
+
+            if modified_location.id != location_id_:
+                raise ValueError("A Location's 'id' cannot be modified")
+
+            self._location_use_case.update(modified_location, **kwargs)
+
+            return HTTPStatus.OK, dumps(JsonTranslator.to_json(modified_location), indent=2)
+
+        @self._rest_controller.register_rest_endpoint(
+            "/api/location/<location_id>/timeline", RESTMethod.GET, MIMEType.JSON, query_params=True
+        )
+        def location_timeline_get_handler(query_params: Dict[str, str], *, location_id: str, **kwargs) -> HandlerResult:
+            if not location_id.startswith("location-"):
+                raise ValueError(f"Cannot parse location id from '{location_id}")
+            location_id_ = JsonTranslator.from_json(location_id, PrefixedUUID)
+
+            supported_filters = {"taggedAll", "taggedAny", "taggedOnly", "taggedNone"}
+            if not supported_filters.issuperset(query_params.keys()):
+                raise ValueError(f"Unsupported filter(s): {', '.join(query_params.keys() - supported_filters)}")
+            filters = {
+                "tagged_all": parse_optional_tag_set_query_param(query_params.get("taggedAll", None)),
+                "tagged_any": parse_optional_tag_set_query_param(query_params.get("taggedAny", None)),
+                "tagged_only": parse_optional_tag_set_query_param(query_params.get("taggedOnly", None)),
+                "tagged_none": parse_optional_tag_set_query_param(query_params.get("taggedNone", None)),
+            }
+
+            timeline = self._timeline_use_case.construct_location_timeline(location_id_, **filters, **kwargs)
+
+            return HTTPStatus.OK, dumps(JsonTranslator.to_json(timeline), indent=2)
+
+
+class TravelersRestRequestHandler:
+    _traveler_use_case: TravelerUseCase
+    _timeline_use_case: TimelineUseCase
+    _rest_controller: RESTController
+
+    def __init__(self, rest_controller: RESTController, traveler_use_case: TravelerUseCase, timeline_use_case: TimelineUseCase) -> None:
+        self._rest_controller = rest_controller
+        self._traveler_use_case = traveler_use_case
+        self._timeline_use_case = timeline_use_case
+
+    def register_routes(self):
```

Comment:
The event routes are done in the init. Decide which pattern to go with and unify it.

Reply:
Improved these to be static registration methods.